ember-client/src/main/scala/org/http4s/ember/client/internal/ClientHelpers.scala

```diff
@@ -101,33 +101,30 @@ private[client] object ClientHelpers {
         .resource
         .drain
 
-    def onNoTimeout(req: Request[F], socket: Socket[F]): Resource[F, Response[F]] =
-      writeRequestToSocket(req, socket, None) >>
-        Parser.Response.parser(maxResponseHeaderSize)(
-          socket.reads(chunkSize, None)
-        )
-
-    def onTimeout(
-        req: Request[F],
-        socket: Socket[F],
-        fin: FiniteDuration): Resource[F, Response[F]] =
-      for {
-        start <- RT.clock.realTime(MILLISECONDS)
-        _ <- writeRequestToSocket(req, socket, Option(fin))
-        timeoutSignal <- Resource.liftF(SignallingRef[F, Boolean](true))
-        sent <- RT.clock.realTime(MILLISECONDS)
-        remains = fin - (sent - start).millis
-        resp <- Parser.Response.parser[F](maxResponseHeaderSize)(
-          readWithTimeout(socket, start, remains, timeoutSignal.get, chunkSize)
-        )
-        _ <- Resource.liftF(timeoutSignal.set(false).void)
-      } yield resp
-
-    def writeRead(req: Request[F]) =
-      timeout match {
-        case t: FiniteDuration => onTimeout(req, requestKeySocket.socket, t)
-        case _ => onNoTimeout(req, requestKeySocket.socket)
+    def writeRead(req: Request[F]) = writeRequestToSocket(req, requestKeySocket.socket, None) >> {
```

Comment:
Assuming the return type is `F[Response[F]]` here but annotation would be nice

Reply:
`Resource[F, Response[F]]`